src/middleware.ts

```diff
@@ -156,6 +156,18 @@ export const combine =
       )
     )
 
+type RemoveMethods<T> = Pick<
```

Comment:
btw, it's always a joy to get headache while working with ts generics :upside_down_face: